[cppcheck] fix invocation & performance checks

Rechi opened the pull request:
## Description
fix invocation of cppcheck
fix cppcheck performance checks

## Motivation and context
fix running `analyze-cppcheck` target on Debian testing

## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

garbear commented on cmake/modules/buildtools/FindCppcheck.cmake:
```diff
@@ -21,7 +21,10 @@ if(CPPCHECK_FOUND)
   # but doesn't support OBJC and OBJCXX for <LANG>_CLANG_TIDY.
-  file(WRITE "${CMAKE_BINARY_DIR}/cppcheck" "case \"$@\" in *.m|*.mm) exit 0; esac\nexec \"${CPPCHECK_EXECUTABLE}\" --enable=performance --quiet --relative-paths=\"${CMAKE_SOURCE_DIR}\" \"$@\"\n")
+  if(CPPCHECK_VERSION VERSION_GREATER_EQUAL 2.14)
+    set(ARGS --check-level=exhaustive)
```
It's not clear from the patch if `ARGS` is used above, which would affect the command. A more verbose name would be better.

Rechi replied:
done

garbear reviewed (approved):
The code changes look good, just a naming suggestion for the cmake stuff.

garbear reviewed (approved):
Thanks for updating the variable name, much more clear and safe now

Rechi committed:
[cppcheck] passedByValue

Rechi committed:
[cppcheck] remove rule-file

- ignored checks are no longer required after 45427e386f
- Debian disabled rules support as it depends on pcre

Rechi committed:
[cppcheck] returnByReference

Rechi committed:
[cppcheck] stlIfStrFind

Rechi committed:
[cppcheck] returnStdMoveLocal

Rechi committed:
[cppcheck] enable exhaustive checking for cppcheck >= 2.14.0
